### PR TITLE
Follow up on the addition of Preview langversion, handle custom diagnostics.

### LINKED ISSUE #33684: Incomplete handling of Preview language version in diagnostics
Look for "8.0" in src\Compilers\CSharp\Test\Semantic\Semantics\NullableReferenceTypesTests.cs. There are a lot of references to C# 8.0 in error messages. For example:
```
// error CS8630: Invalid 'NullableContextOptions' value: 'Enable' for C# 7.3. Please use language version 8.0 or greater.
Diagnostic(ErrorCode.ERR_NullableOptionNotAvailable).WithArguments("NullableContextOptions", "Enable", "7.3", "8.0").WithLocation(1, 1)
```

Expected:
The message to suggest to use "use the 'preview' language version" suggestion instead.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #33684.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -4870,26 +4870,26 @@ public void NullableOption()
         {
             var comp = CreateCompilation("", options: WithNonNullTypes(NullableContextOptions.Enable), parseOptions: TestOptions.Regular7_3);
             comp.VerifyDiagnostics(
-                // error CS8630: Invalid 'NullableContextOptions' value: 'Enable' for C# 7.3. Please use language version 8.0 or greater.
-                Diagnostic(ErrorCode.ERR_NullableOptionNotAvailable).WithArguments("NullableContextOptions", "Enable", "7.3", "8.0").WithLocation(1, 1)
+                // error CS8630: Invalid 'NullableContextOptions' value: 'Enable' for C# 7.3. Please use language version preview or greater.
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: >Please use language version preview or greater. [](start = 94, length = 47)

nit: it's not as clear as before that the sentence contains a value. Consider adding single quotes: `Please use language version 'preview' or greater.`

File: src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs [hunk 2]
```diff
@@ -57,9 +57,9 @@ public void TestInterpolatedVerbatimString()
         public void TestAltInterpolatedVerbatimString_CSharp73()
         {
             UsingExpression(@"@$""hello""", TestOptions.Regular7_3,
-                // (1,1): error CS8401: To use '@$' instead of '$@' for an interpolated verbatim string, please use language version 8.0 or greater.
+                // (1,1): error CS8401: To use '@$' instead of '$@' for an interpolated verbatim string, please use language version preview or greater.
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Syntax/Parsing/ExpressionParsingTests.cs]: >please use language version preview or greater. [](start = 105, length = 47)

same suggestion here and possible some other messages that use RequiredLanguageVersion

**Comment by AlekseyTs** [comment]
@agocke Please review

**Comment by AlekseyTs** [comment]
@agocke Please review

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs (hunk 1 above)
Comment (agocke) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: I agree with this

**Pull request merged**